lib/in_stock/application.ex

```diff
@@ -26,6 +25,33 @@ defmodule InStock.Application do
     Supervisor.start_link(children, opts)
   end
 
+  # Returns a child spec for an ad-hoc supervisor that will start the dynamic
+  # supervisor CheckerSupervisor and then start its initial children
+  #
+  # References:
+  # - https://elixirforum.com/t/understanding-dynamicsupervisor-no-initial-children/14938/2
+  # - https://github.com/slashdotdash/til/blob/master/elixir/dynamic-supervisor-start-children.md
+  defp checker_child_spec() do
+    checker_children = [
+      InStock.Checker.CheckerSupervisor,
+      {Task, &InStock.Checker.CheckerSupervisor.spawn_checkers/0}
+    ]
```

Comment:
Sets up the children for the adhoc sup: the dynamic sup and then a task that runs its spawn_checkers

The task will run after the dynamic sup is started and ready